java/arcs/sdk/Particle.kt

```diff
@@ -11,8 +11,33 @@
 
 package arcs.sdk
 
+/**
+ * Interface used by [ArcHost]s to interact dynamically with code-generated [Handle] fields
+ * used by [Particle]s.
+ */
+interface HandleHolder {
+    val map: MutableMap<String, Handle>
+    val entitySpecs: MutableMap<String, EntitySpec<out Entity>>
```

Comment:
These don't need to be mutable, if you follow some of my other suggestions in this file.